src/runtime/pkg/katautils/config.go

```diff
@@ -1009,15 +1009,16 @@ func newClhHypervisorConfig(h hypervisor) (vc.HypervisorConfig, error) {
 		return vc.HypervisorConfig{}, err
 	}
 
-	if sharedFS != config.VirtioFS && sharedFS != config.VirtioFSNydus {
-		return vc.HypervisorConfig{}, errors.New("clh only support virtio-fs or virtio-fs-nydus")
-	}
-
-	if h.VirtioFSDaemon == "" {
+	if (sharedFS == config.VirtioFS || sharedFS != config.VirtioFSNydus) && h.VirtioFSDaemon == "" {
 		return vc.HypervisorConfig{},
 			fmt.Errorf("cannot enable %s without daemon path in configuration file", sharedFS)
 	}
 
+
+	if sharedFS == config.Virtio9P {
```

Comment:
The change you're making here is analogous to going from an allow-list to a deny-list.

I think the consensus is that an allow-list is better though.

In this particular case, suppose we add a new Fs to kata that clh doesn't support; with a deny-list, we would wrongly accept it. With an allow-list, we would reject it. I think the latter is preferable because if it is meant to be allowed, someone will notice it and just add it here. Allowing something that shouldn't be allowed is harder to catch.

Reply:
Sure, I will change the check back to how it was, and also re-write the check above.